### PR TITLE
Reconcile the "easy" System\Reflection .cs files with their CoreRT ve…

### PR DESCRIPTION
**Pull request by AtsushiKan**
…rsions.

Files to be moved the shared partition (after CoreRT side is in)

AmbiguousMatchException.cs
AssemblyAttributes.cs
AssemblyContentType.cs
AssemblyNameFlags.cs
Binder.cs
BindingFlags.cs
CallingConventions.cs
CustomAttributeFormatException.cs
DefaultMemberAttribute.cs
EventAttributes.cs
ExceptionHandlingClauseOptions.cs
FieldAttributes.cs
GenericParameterAttributes.cs
ICustomAttributeProvider.cs
ImageFileMachine.cs
InterfaceMapping.cs
IntrospectionExtensions.cs
InvalidFilterCriteriaException.cs
IReflectableType.cs
ManifestResourceInfo.cs
MemberFilter.cs
MemberTypes.cs
MethodAttributes.cs
MethodImplAttributes.cs
Missing.cs
ModuleResolveEventHandler.cs
ParameterAttributes.cs
ParameterModifier.cs
PortableExecutableKinds.cs
ProcessorArchitecture.cs
PropertyAttributes.cs
ReflectionContext.cs
ReflectionTypeLoadException.cs
ResourceAttributes.cs
ResourceLocation.cs
StrongNameKeyPair.cs
TargetException.cs
TargetInvocationException.cs
TargetParameterCountException.cs
TypeAttributes.cs
TypeFilter.cs

### CODE REVIEW

**Comment by AtsushiKan** [comment]
cc @jkotas

**Comment by AtsushiKan** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test

File: src/mscorlib/src/System/Reflection/IntrospectionExtensions.cs [hunk 1]
```diff
@@ -2,28 +2,27 @@
...
+            IReflectableType reflectableType = type as IReflectableType;
+            if (reflectableType != null)
+                return reflectableType.GetTypeInfo();
+
             if (type == null)
                 throw new ArgumentNullException(nameof(type));
 
-            var rcType = (IReflectableType)type;
-            return rcType.GetTypeInfo();
+            // This is bizarre but compatible with the desktop which casts "type" to IReflectableType without checking and
```
Comment (jkotas) [inline src/mscorlib/src/System/Reflection/IntrospectionExtensions.cs]: I think that the CoreCLR implementation is much easier to follow, and it is also smaller - will be better candidate for inlining, etc.

(It is fine to add comment to it that the cast throws InvalidCastException.)

File: src/mscorlib/src/System/Reflection/ExceptionHandlingClause.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+{
+    public class ExceptionHandlingClause
+    {
+        #region costructor
+        // This class can only be created from inside the EE.
+        protected ExceptionHandlingClause() { }
+        #endregion
+
+        #region Private Data Members
+        private MethodBody m_methodBody;
```
Comment (jkotas) [inline src/mscorlib/src/System/Reflection/ExceptionHandlingClause.cs]: It would be nice to remove the `m` prefixes...

Comment (AtsushiKan) [inline src/mscorlib/src/System/Reflection/ExceptionHandlingClause.cs]: Maybe on some other occasion. The fields are accessed from native code so renaming them means adding more diffs than just this one file.

**Comment by AtsushiKan** [comment]
Incorporated the suggested changes on the CoreRt change - they're small enough that they should pass through the mirrorbot easily.

**Pull request merged**